### PR TITLE
Add default endpoint and custom endpoint support to AMC

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

**Comment by pakrym** [comment]
cc @richardpark-msft

File: sdk/monitor/Azure.Monitor.Query/api/Azure.Monitor.Query.netstandard2.0.cs [hunk 1]
```diff
@@ -20,6 +22,7 @@ public partial class LogsQueryClient
     public partial class LogsQueryClientOptions : Azure.Core.ClientOptions
     {
         public LogsQueryClientOptions(Azure.Monitor.Query.LogsQueryClientOptions.ServiceVersion version = Azure.Monitor.Query.LogsQueryClientOptions.ServiceVersion.V1) { }
+        public string AuthenticationScope { get { throw null; } set { } }
```
Comment (richardpark-msft) [inline sdk/monitor/Azure.Monitor.Query/api/Azure.Monitor.Query.netstandard2.0.cs:25]: Is this auto-generated code? I'm kind of surprised to see a throw inside of the property here.

Comment (christothes) [reply sdk/monitor/Azure.Monitor.Query/api/Azure.Monitor.Query.netstandard2.0.cs:25]: It's the API export

File: sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientClientTests.cs [hunk 2]
```diff
@@ -76,5 +78,54 @@ public async Task QueryBatchHandledInvalidResponse()
...
+        public async Task UsesDefaultAuthScope(string scope, string expectedScope)
+        {
+            var mockTransport = MockTransport.FromMessageCallback(message =>
+            {
+                var mockResponse = new MockResponse(200);
+                mockResponse.SetContent("{\"tables\":[]}");
+                return mockResponse;
+            });
+
+            Mock<MockCredential> mock = new() { CallBase = true };
```
Comment (christothes) [inline sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientClientTests.cs:114]: FYI - [this change](https://github.com/Azure/azure-sdk-for-net/blob/59f2c7dae9202df896c0e7077d9d154198a88260/sdk/core/Azure.Core.TestFramework/src/MockCredential.cs) (yet to merge) makes it easy to setup callbacks for calls to GetToken.

**Review by christothes — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.Query/api/Azure.Monitor.Query.netstandard2.0.cs (hunk 1 above)
Comment (pakrym) [reply sdk/monitor/Azure.Monitor.Query/api/Azure.Monitor.Query.netstandard2.0.cs:25]: Yeah, that's the exported API. The API exporter produces something that can be compiled so it replaces method bodies with `throw null;`.

Comment (richardpark-msft) [reply sdk/monitor/Azure.Monitor.Query/api/Azure.Monitor.Query.netstandard2.0.cs:25]: Ah, that is interesting. Our equivalent of this file is just a api.md file :)

**Pull request merged**